diff --git a/CMakeLists.txt b/CMakeLists.txt
index 228b4a5..343651e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,10 +30,16 @@ if (Qt5Core_FOUND)
   set(CMAKE_LIBATTICA_VERSION_MAJOR 1)
   set(CMAKE_LIBATTICA_VERSION_MINOR 0)
   set(CMAKE_LIBATTICA_VERSION_PATCH 0)
+  set(ATTICA_LIB_SONAME KF5Attica)
+  set(ATTICA_LIB_SONAME_CAMEL LibKF5Attica)
+  set(ATTICA_LIB_NAMESPACE KF5)
 else()
   set(CMAKE_LIBATTICA_VERSION_MAJOR 0)
   set(CMAKE_LIBATTICA_VERSION_MINOR 4)
   set(CMAKE_LIBATTICA_VERSION_PATCH 2)
+  set(ATTICA_LIB_SONAME attica)
+  set(ATTICA_LIB_SONAME_CAMEL LibAttica)
+  set(ATTICA_LIB_NAMESPACE LibAttica)
 endif()
 set(CMAKE_LIBATTICA_VERSION_STRING  "${CMAKE_LIBATTICA_VERSION_MAJOR}.${CMAKE_LIBATTICA_VERSION_MINOR}.${CMAKE_LIBATTICA_VERSION_PATCH}")
 
@@ -59,23 +65,23 @@ endif (ATTICA_ENABLE_TESTS)
 add_subdirectory(lib)
 
 # Write LibAtticaConfigVersion.cmake
-write_basic_package_version_file(${attica_BINARY_DIR}/LibAtticaConfigVersion.cmake
+write_basic_package_version_file(${attica_BINARY_DIR}/${ATTICA_LIB_SONAME_CAMEL}ConfigVersion.cmake
                                  VERSION ${CMAKE_LIBATTICA_VERSION_STRING}
                                  COMPATIBILITY SameMajorVersion
                                 )
 
 # Create the CMake Config files
 configure_package_config_file(LibAtticaConfig.cmake.in
-                              ${attica_BINARY_DIR}/LibAtticaConfig.cmake
-                              INSTALL_DESTINATION ${LIB_DESTINATION}/cmake/LibAttica
+                              ${attica_BINARY_DIR}/${ATTICA_LIB_SONAME_CAMEL}Config.cmake
+                              INSTALL_DESTINATION ${LIB_DESTINATION}/cmake/${ATTICA_LIB_SONAME_CAMEL}
                               PATH_VARS LIB_DESTINATION INCLUDE_DESTINATION
 )
 
-install(FILES ${attica_BINARY_DIR}/LibAtticaConfig.cmake
-              ${attica_BINARY_DIR}/LibAtticaConfigVersion.cmake
-        DESTINATION ${LIB_DESTINATION}/cmake/LibAttica
+install(FILES ${attica_BINARY_DIR}/${ATTICA_LIB_SONAME_CAMEL}Config.cmake
+              ${attica_BINARY_DIR}/${ATTICA_LIB_SONAME_CAMEL}ConfigVersion.cmake
+        DESTINATION ${LIB_DESTINATION}/cmake/${ATTICA_LIB_SONAME_CAMEL}
        )
 
 install(EXPORT LibAtticaExport FILE LibAtticaTargets.cmake
-        DESTINATION ${LIB_DESTINATION}/cmake/LibAttica
+        DESTINATION ${LIB_DESTINATION}/cmake/${ATTICA_LIB_SONAME_CAMEL}
         NAMESPACE LibAttica::)
diff --git a/LibAtticaConfig.cmake.in b/LibAtticaConfig.cmake.in
index 51d1804..3049c4a 100644
--- a/LibAtticaConfig.cmake.in
+++ b/LibAtticaConfig.cmake.in
@@ -22,7 +22,7 @@ set_and_check(LibAttica_LIBRARY_DIRS "@PACKAGE_LIB_DESTINATION@")
 ########## The LibAttica libraries ##########
 # Load the exported targets.
 include("${CMAKE_CURRENT_LIST_DIR}/LibAtticaTargets.cmake")
-set(LibAttica_LIBRARIES        LibAttica::attica)
+set(LibAttica_LIBRARIES        @ATTICA_LIB_NAMESPACE@::@ATTICA_LIB_SONAME@)
 ###########################################
 
 # The following variables are kept for compatibility purpose
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 3585523..6639c00 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -89,14 +89,14 @@ set(ocsclient_SRCS
 qt4_automoc(${ocsclient_SRCS})
 
 if (ATTICA_STATIC_BUILD)
-    add_library(attica STATIC ${ocsclient_SRCS})
-    set_target_properties(attica PROPERTIES
+    add_library(${ATTICA_LIB_SONAME} STATIC ${ocsclient_SRCS})
+    set_target_properties(${ATTICA_LIB_SONAME} PROPERTIES
         VERSION ${CMAKE_LIBATTICA_VERSION_STRING}
         SOVERSION ${CMAKE_LIBATTICA_VERSION_MAJOR}
     )
 else (ATTICA_STATIC_BUILD)
-    add_library(attica SHARED ${ocsclient_SRCS})
-    set_target_properties(attica PROPERTIES
+    add_library(${ATTICA_LIB_SONAME} SHARED ${ocsclient_SRCS})
+    set_target_properties(${ATTICA_LIB_SONAME} PROPERTIES
         VERSION ${CMAKE_LIBATTICA_VERSION_STRING}
         SOVERSION ${CMAKE_LIBATTICA_VERSION_MAJOR}.${CMAKE_LIBATTICA_VERSION_MINOR}
         LINK_INTERFACE_LIBRARIES ""
@@ -105,13 +105,13 @@ else (ATTICA_STATIC_BUILD)
 endif(ATTICA_STATIC_BUILD)
 
 if(Qt5Core_FOUND)
-  set_property(TARGET attica APPEND PROPERTY COMPILE_DEFINITIONS QT_DISABLE_DEPRECATED_BEFORE=0)
+  set_property(TARGET ${ATTICA_LIB_SONAME} APPEND PROPERTY COMPILE_DEFINITIONS QT_DISABLE_DEPRECATED_BEFORE=0)
 endif()
 
-target_link_libraries(attica ${QT_QTCORE_LIBRARY} ${QT_QTNETWORK_LIBRARY})
-target_include_directories(attica INTERFACE $<INSTALL_INTERFACE:${INCLUDE_DESTINATION}>)
-
-install(TARGETS attica EXPORT LibAtticaExport ARCHIVE DESTINATION ${LIB_DESTINATION}
+target_link_libraries(${ATTICA_LIB_SONAME} ${QT_QTCORE_LIBRARY} ${QT_QTNETWORK_LIBRARY})
+target_include_directories(${ATTICA_LIB_SONAME} INTERFACE $<INSTALL_INTERFACE:${INCLUDE_DESTINATION}>)
+ 
+install(TARGETS ${ATTICA_LIB_SONAME} EXPORT LibAtticaExport ARCHIVE DESTINATION ${LIB_DESTINATION}
                                LIBRARY DESTINATION ${LIB_DESTINATION}
                                RUNTIME DESTINATION bin)
 
@@ -160,15 +160,15 @@ install(FILES
     publisherfield.h
     topic.h
     ${CMAKE_CURRENT_BINARY_DIR}/version.h
-    DESTINATION ${INCLUDE_DESTINATION}/attica COMPONENT Devel
+    DESTINATION ${INCLUDE_DESTINATION}/${ATTICA_LIB_SONAME} COMPONENT Devel
 )
 
 # Install package config file
 if(NOT WIN32)
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/libattica.pc.cmake
-        ${CMAKE_CURRENT_BINARY_DIR}/libattica.pc
+        ${CMAKE_CURRENT_BINARY_DIR}/lib${ATTICA_LIB_SONAME}.pc
     )
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libattica.pc
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/lib${ATTICA_LIB_SONAME}.pc
         DESTINATION ${LIB_DESTINATION}/pkgconfig
     )
 endif(NOT WIN32)
diff --git a/lib/cmake/libattica.pc.cmake b/lib/cmake/libattica.pc.cmake
index 117f10d..728ce20 100644
--- a/lib/cmake/libattica.pc.cmake
+++ b/lib/cmake/libattica.pc.cmake
@@ -3,10 +3,9 @@ exec_prefix=${CMAKE_INSTALL_PREFIX}/bin
 libdir=${LIB_DESTINATION}
 includedir=${INCLUDE_DESTINATION}
 
-Name: libattica
+Name: lib${ATTICA_LIB_SONAME}
 Description: Qt library to access Open Collaboration Services
 #Requires:
 Version: ${CMAKE_LIBATTICA_VERSION_MAJOR}.${CMAKE_LIBATTICA_VERSION_MINOR}.${CMAKE_LIBATTICA_VERSION_PATCH}
-Libs: -L${LIB_DESTINATION} -lattica
+Libs: -L${LIB_DESTINATION} -l${ATTICA_LIB_SONAME}
 Cflags: -I${INCLUDE_DESTINATION}
-
diff --git a/lib/tests/CMakeLists.txt b/lib/tests/CMakeLists.txt
index 4439d28..9e2d901 100644
--- a/lib/tests/CMakeLists.txt
+++ b/lib/tests/CMakeLists.txt
@@ -3,7 +3,7 @@ macro(ATTICA_TESTS)
         qt4_generate_moc(${_testname}.cpp ${CMAKE_CURRENT_BINARY_DIR}/${_testname}.moc)
         include_directories(${attica_BINARY_DIR}/lib)
         add_executable(${_testname} ${_testname}.cpp ${_testname}.moc)
-        target_link_libraries(${_testname} ${QT_QTTEST_LIBRARY} ${QT_QTCORE_LIBRARY} attica)
+        target_link_libraries(${_testname} ${QT_QTTEST_LIBRARY} ${QT_QTCORE_LIBRARY} ${ATTICA_LIB_SONAME})
         add_test(${_testname} ${CMAKE_CURRENT_BINARY_DIR}/${_testname})
     endforeach(_testname)
 endmacro(ATTICA_TESTS)
diff --git a/lib/tests/projecttest/CMakeLists.txt b/lib/tests/projecttest/CMakeLists.txt
index 34ab60a..8d749a7 100644
--- a/lib/tests/projecttest/CMakeLists.txt
+++ b/lib/tests/projecttest/CMakeLists.txt
@@ -24,4 +24,4 @@ qt4_automoc(${projecttest_SRCS} ${editproject_SRCS})
 
 add_executable(projecttest ${projecttest_SRCS} ${editproject_UI_H})
 
-target_link_libraries(projecttest ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} attica)
+target_link_libraries(projecttest ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${ATTICA_LIB_SONAME})
