Description: set LIB_DESTINATION to multiarch library path
Author: Daniel Schaal <farbing@web.de>
Author: Pino Toscano <pino@debian.org>
Last-Update: 2013-06-24
Forwarded: no

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,7 @@ else()
   set(QT_MIN_VERSION "4.7.0")
   find_package(Qt4 REQUIRED)
   set(LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)" )
-  set(LIB_DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" CACHE STRING "Library directory name" FORCE)
+  set(LIB_DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_LIBRARY_ARCHITECTURE}" CACHE STRING "Library directory name" FORCE)
 endif()
 
 if (Qt5Core_FOUND)
